stdweb-derive/src/lib.rs

```diff
@@ -69,6 +70,18 @@ pub fn derive_reference_type( input: TokenStream ) -> TokenStream {
                         panic!( "The value of '#[reference(instance_of = ...)]' is not a string!" );
                     }
                 },
+                syn::NestedMeta::Meta( syn::Meta::NameValue( ref meta ) ) if meta.ident == "constraint" => {
+                    if let syn::Lit::Str( ref str ) = meta.lit {
+                        let val = str.value();
+                        let parts : Vec<&str> = val.splitn(2,"=").collect();
+                        if parts.len() != 2 {
+                            panic!("The value of '#[reference(..., constraint = ...)]' must be in the form of \"js_field=value\"" );
+                        }
+                        constraints.push(( parts[0].to_owned(), parts[1].to_owned()) );
```

Comment:
For now I think it'd be better to be more conservative and make it accept only one constraint, and error out in case of a duplicate. (So `Option` instead of a `Vec`.)

Reply:
Either way is fine by me. Will make the change